Source/FullScreenPokemon.ts

```diff
@@ -3149,6 +3250,50 @@ module FullScreenPokemon {
                 }
             }
 
+            console.log(other);
+            console.log("jumping character's top " + thing.top);
+            for (let border of (<ILedge>other).border[thing.direction]) {
+                switch (thing.direction) {
+                    case 0:
+                        if (border.left === thing.left) {
+                            thing.FSP.animateCharacterStopWalking(thing);
+                            thing.FSP.centerMapScreen();
```

Comment:
Why?

In an ideal world the ledge should act just like a normal 8x8 solid so you shouldn't have to interact with scrolling logic at all.